### PR TITLE
added mikrotik webauth support

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Adds support to Mikrotik webauth

## WARNING: This is not fully tested as I don't have access to the equipment

# Impacts
Mikrotik

# Delete branch after merge
YES

# NEWS file entries
## New Features
* Added support for web authentication for the Mikrotik access points

### CODE REVIEW

**Comment by julsemaan** [comment]
@dwlfrth, I developped this under a pre-webauth-refactor version

Could you make sure I followed the appropriate steps to have it working under the current version

**Comment by extrafu** [comment]
Bump - @dwlfrth what's up on this?

File: lib/pf/Switch/Mikrotik.pm [hunk 1]
```diff
@@ -48,6 +49,70 @@ sub supportsWirelessMacAuth { return $TRUE; }
...
+    my $logger = $self->logger;
+
+    # Using a hash to contain external portal parameters
+    my %params = ();
+
+    %params = (
+        switch_id       => $req->param('ap-id'),
+        client_mac      => clean_mac($req->param('mac')),
+        client_ip       => $req->param('ip'),
+        ssid            => undef,
```
Comment (dwlfrth) [inline lib/pf/Switch/Mikrotik.pm]: No need to specify undef parameters since they are undef by default in lib/pf/web/externalportal.pm

**Review by dwlfrth — CHANGES_REQUESTED** [review]
Other than the one comment, LGTM

**Comment by julsemaan** [comment]
fixaide

**Pull request merged**